aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -0,0 +1,144 @@
+        self.headers["Access-Control-Allow-Methods"] = method
+        self.headers["Access-Control-Allow-Credentials"] = "true"
+
+    def add_cache_control(self, cache_control: str):
+        self.headers["Cache-Control"] = cache_control if self.status_code == 200 else "no-cache"
+
+    def compress(self):
+        self.headers["Content-Encoding"] = "gzip"
+        if isinstance(self.body, str):
+            self.body = bytes(self.body, "utf-8")
+        gzip = zlib.compressobj(9, zlib.DEFLATED, zlib.MAX_WBITS | 16)
+        self.body = gzip.compress(self.body) + gzip.flush()
+
+    def to_dict(self):
+        if isinstance(self.body, bytes):
+            self.base64_encoded = True
+            self.body = base64.b64encode(self.body).decode()
+        return {
+            "statusCode": self.status_code,
+            "headers": self.headers,
+            "body": self.body,
+            "isBase64Encoded": self.base64_encoded,
+        }
+
+
+class ApiGatewayResolver:
+    current_event: BaseProxyEvent
+    lambda_context: LambdaContext
+
+    def __init__(self, proxy_type: Enum = ProxyEventType.http_api_v1):
+        self._proxy_type = proxy_type
+        self._routes: List[Route] = []
+
+    def get(self, rule: str, cors: bool = False, compress: bool = False, cache_control: str = None):
+        return self.route(rule, "GET", cors, compress, cache_control)
+
+    def post(self, rule: str, cors: bool = False, compress: bool = False, cache_control: str = None):
+        return self.route(rule, "POST", cors, compress, cache_control)
+
+    def put(self, rule: str, cors: bool = False, compress: bool = False, cache_control: str = None):
+        return self.route(rule, "PUT", cors, compress, cache_control)
+
+    def delete(self, rule: str, cors: bool = False, compress: bool = False, cache_control: str = None):
+        return self.route(rule, "DELETE", cors, compress, cache_control)
+
+    def patch(self, rule: str, cors: bool = False, compress: bool = False, cache_control: str = None):
+        return self.route(rule, "PATCH", cors, compress, cache_control)
+
+    def route(self, rule: str, method: str, cors: bool = False, compress: bool = False, cache_control: str = None):
+        def register_resolver(func: Callable):
+            self._routes.append(Route(method, self._build_rule_pattern(rule), func, cors, compress, cache_control))
+            return func
+
+        return register_resolver
+
+    def resolve(self, event, context) -> Dict[str, Any]:
+        self.current_event = self._as_data_class(event)
+        self.lambda_context = context
+        route, args = self._find_route(self.current_event.http_method, self.current_event.path)
+        result = route.func(**args)
+
+        if isinstance(result, Response):
+            response = result
+        elif isinstance(result, dict):
+            response = Response(status_code=200, content_type="application/json", body=json.dumps(result))
```

Comment:
do you want to default `json.dumps` to `json.dumps(result, separators=(',', ':'), indent=None, sort_keys=True)` for consistent minimized outputs?

also would it make sense to set `content_type="application/json; charset=utf-8"`?

Reply:
@jplock true, it could be more concise.

Reply:
@jplock - seems like UTF-8 is the default encoding for JSON anyway: https://www.ietf.org/rfc/rfc4627.txt

Reply:
We'll have to revert the `separator` optimization - This breaks people's unit tests, for example:

```python
assert ret["body"] == json.dumps({"message": "hello universe"})
```

```python
FAILED tests/unit/test_hello_world_function.py::test_lambda_handler - assert '{"message":"hello universe"}' == '{"message": "hello universe"}'
```

I'll have dinner and release a patch version with it